frontend/web/components/modals/ConfirmDeleteRole.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { FC, useEffect } from 'react'
+import Button from 'components/base/forms/Button'
+import Utils from 'common/utils/utils'
+import { useDeleteRoleMutation } from 'common/services/useRole'
+import ModalHR from './ModalHR'
+import ErrorMessage from 'components/ErrorMessage'
+
+type ConfirmDeleteRoleType = {
+  role: Role
+  onComplete?: () => void
+}
+const ConfirmDeleteRole: FC<ConfirmDeleteRoleType> = ({ onComplete, role }) => {
+  const [deleteRole, { isError, isSuccess: deleted }] = useDeleteRoleMutation()
+
+  useEffect(() => {
+    if (deleted) {
+      onComplete?.()
+      closeModal()
+    }
+  }, [deleted, onComplete])
+
+  const ModalBody: FC<ConfirmDeleteRoleType> = ({ role }) => {
```

Comment:
Does this need to be defined in the render function? Also, why is it even a separate component ?

Reply:
It is not necessary to have it there. I placed it in a separate component to keep it independent if I needed to make more significant changes, but that won't be the case. I already moved it into the return.